### PR TITLE
[connectors] Implement permission setting in Zendesk

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Close [#1505](https://github.com/dust-tt/tasks/issues/1505)
- Close [#1522](https://github.com/dust-tt/tasks/issues/1522)

## Deploy Plan

- Merge
- Run migration_28
- Deploy connectors

### CODE REVIEW

File: connectors/src/connectors/zendesk/index.ts [hunk 1]
```diff
@@ -123,8 +144,145 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
...
+      );
+      return new Err(new Error("ZendeskConfiguration not found"));
+    }
+    const subdomain = zendeskConfiguration.subdomain;
+
+    const toBeSignaledBrandIds = new Set<number>();
+    const toBeSignaledTicketsIds = new Set<number>();
+    const toBeSignaledHelpCenterIds = new Set<number>();
+    const toBeSignaledCategoryIds = new Set<number>();
+    try {
```
Comment (fontanierh) [inline connectors/src/connectors/zendesk/index.ts:173]: IMHO we shouldn't have this "catch all" try/catch. If the error is an unexpected exception, you should let it throw. The endpoint will 500 which is what we want.
We return `Err` when the error is "expected".

File: connectors/src/connectors/zendesk/lib/help_center_permissions.ts [hunk 2]
```diff
@@ -25,6 +25,170 @@ import {
...
+  const category = await ZendeskCategoryResource.fetchByCategoryId({
+    connectorId,
+    categoryId: categoryId,
+  });
+  if (!category) {
+    logger.error(
+      { categoryId: categoryId },
+      "[Zendesk] Category not found, could not revoke sync."
+    );
+    return null;
```
Comment (fontanierh) [inline connectors/src/connectors/zendesk/lib/help_center_permissions.ts:185]: you don't want to throw for this one ?

**Review by fontanierh — APPROVED** [review]
Looks great, I don't have full context on zendesk but definitely looks reasonable 👍

File: connectors/src/connectors/zendesk/lib/help_center_permissions.ts (hunk 2 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/lib/help_center_permissions.ts:185]: figured that if we were to revoke permissions for smth that didn't exist anymore we should carry on and try to process the other permissions

Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/lib/help_center_permissions.ts:185]: I do realize that's not consistent with the behavior when allowing permissions, I'm a bit undecided on that one 🤷

**Review by spolu — APPROVED** [review]
LGTM

Both approach are defensible on the return null. We can def go one way or another for now.

File: connectors/src/connectors/zendesk/lib/help_center_permissions.ts (hunk 2 above)
Comment (fontanierh) [reply connectors/src/connectors/zendesk/lib/help_center_permissions.ts:185]: Fair !

**Comment by aubin-tchoi** [comment]
Final decision regarding the `return null` vs `throw`: `return null` everywhere (allow and revoke)

**Pull request merged**